app/src/test/java/com/codesoom/assignment/application/ProductServiceTest.java

```diff
@@ -0,0 +1,194 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+// Repository를 주입하여 테스트하기
+// 1.@DataJpaTest  //jpa를 직접 사용하여 테스트할 수 있다.
+// 2. TaskRepository taskRepository = mock(TaskRepository.class)
+@DisplayName("ProductService 클래스")
+class ProductServiceTest {
+    private ProductService productService;
+    private ProductRepository productRepository;
+
+    @BeforeEach
+    void setUp() {
+        productRepository = mock(ProductRepository.class);
+        productService = new ProductService(productRepository);
+
+        List<Product> products = new ArrayList<>();
+
+        Product product = Product.builder()
+                .name("product Name")
+                .maker("maker")
+                .build();
+        // 기존 week3에서는 service의 create가 올바르게 작동할 때만 정상 작동할 수 있었다.
+        // 이제부터는 repository에만 의존하면 된다.
+        products.add(product);
+
+        //given
+        given(productRepository.findAll()).willReturn(products);
+        given(productRepository.findById(1L)).willReturn(Optional.of(product));
+        given(productRepository.findById(100L)).willReturn(Optional.empty());
+    }
+/*
+    void setUpSaveProduct() {
+        given(productRepository.save(any(Product.class))).will(invocation -> {
+            Product product = invocation.getArgument(0);
+            product.setId(1L);
+            return product;
+        });
+    }
+*/
+    @Nested
+    @DisplayName("getProducts 메소드")
+    class describe_getProducts{
+
+        @Test
+        @DisplayName("List<Product>를 반환한다.")
+        void it_returns_list(){
+            List<Product> products = productService.getProducts();
+            verify(productRepository).findAll();
+            assertThat(products).isInstanceOf(List.class);
```

Comment:
```suggestion
            List<Product> products = productService.getProducts();

            verify(productRepository).findAll();

            assertThat(products).isInstanceOf(List.class);
```

적절히 빈 줄로 나눠주는게 좋을 것 같습니다. 왜냐하면 각 줄이 다른 일을 하고 있는 것 같아요. 우리가 일반적인 글쓰기를 할 때 문단을 나누는 것처럼 코드도 문단을 잘 나누면 더 읽기 좋은 코드가 됩니다

Reply:
반영하였습니다.
ProductControllerApiTest.java를 커밋하면서 개행을 많이 적용해 보았는데요.
해당 코드는 과도한 개행인지 혹은 적당한 개행인지 확인해주실 수 있으신가요?